frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/font_family_setting.dart

```diff
@@ -109,35 +111,61 @@ class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
         .replaceAllMapped(camelCase, (m) => ' ${m.group(0)}');
   }
 
+  bool _isTextOverflowing(String text, TextStyle textStyle, double maxWidth) {
+    final textPainter = TextPainter(
+      text: TextSpan(text: text, style: textStyle),
+      maxLines: 1,
+      textDirection: ui.TextDirection.ltr,
+    )..layout(maxWidth: maxWidth);
+
+    return textPainter.didExceedMaxLines;
+  }
+
   Widget _fontFamilyItemButton(BuildContext context, TextStyle style) {
     final buttonFontFamily = parseFontFamilyName(style.fontFamily!);
-    return SizedBox(
-      key: UniqueKey(),
-      height: 32,
-      child: FlowyButton(
-        key: Key(buttonFontFamily),
-        onHover: (_) => FocusScope.of(context).unfocus(),
-        text: FlowyText.medium(
-          parseFontFamilyName(style.fontFamily!),
-          fontFamily: style.fontFamily!,
+
+    final bool isOverFlown = _isTextOverflowing(
+      buttonFontFamily,
+      Theme.of(context).textTheme.bodyMedium!.copyWith(
+            fontWeight: FontWeight.w500,
+            fontFamily: buttonFontFamily,
+          ),
+      110,
+    );
+
+    return Tooltip(
+      message: isOverFlown ? buttonFontFamily : "",
+      margin: const EdgeInsets.only(right: 220),
```

Comment:
I feel it looks better if we remove the margin(default tooltip margin instead). Closer to the original font preview. 
It did cover the next line, but if we moved the cursor around, it would disappear anyway. 

What do you think? @AslamThachapalli @MayurSMahajan @annieappflowy 
<img width="646" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/e0bcbabf-3418-43c0-9828-430e7e783b05">

Reply:
Also that `220` is another magic number.

Reply:
@hyj1204 I agree.

Reply:
Ok, will correct this. 
What my thought was even though there is a tooltip the user will be able to see the below text option. 
But as you pointed out, it will anyway disappear when the cursor is moved around.

@Xazin Yeah, actually I tried with different margins and 220 looked good, so fixed it. Will revert to the default way :)